cypress/integration/tests/administrator/custom-migration-targets/crud.test.ts

```diff
@@ -33,31 +39,100 @@ describe("Custom Migration Targets CRUD operations", { tags: ["@tier1", "@dc"] }
         cy.intercept("DELETE", "/hub/rulebundles*/*").as("deleteRule");
     });
 
-    it("Custom Migration Targets CRUD", { tags: ["@tier1", "@dc"] }, function () {
-        CustomMigrationTarget.open();
-        const target = new CustomMigrationTarget(
-            data.getRandomWord(8),
-            data.getDescription(),
-            "img/cloud.png",
-            "xml/javax-package-custom-target.windup.xml"
-        );
-        target.create();
-        cy.wait("@postRule");
-        cy.contains(CustomMigrationTargetView.takeMeThereNotification).click();
-        cy.get("article", { timeout: 12 * SEC }).should("contain", target.name);
-
-        const newName = data.getRandomWord(8);
-        target.edit({
-            name: newName,
-            rulesetPath: "xml/javax-package-custom.windup.xml",
-        });
-        cy.wait("@putRule");
-        cy.get("article", { timeout: 12 * SEC }).should("contain", newName);
-        target.name = newName;
-        target.rulesetPath = "xml/javax-package-custom.windup.xml";
-
-        target.delete();
-        cy.wait("@deleteRule");
-        cy.get("article", { timeout: 12 * SEC }).should("not.contain", target.name);
-    });
+    it(
+        "Custom Migration Targets CRUD with manual upload",
+        { tags: ["@tier1", "@dc"] },
+        function () {
+            CustomMigrationTarget.open();
+            const target = new CustomMigrationTarget(
+                data.getRandomWord(8),
+                data.getDescription(),
+                "img/cloud.png",
+                {
+                    type: CustomMigrationTargetOriginType.Manual,
+                    rulesetPath: "xml/javax-package-custom-target.windup.xml",
+                }
+            );
+            target.create();
+            cy.wait("@postRule");
+            cy.contains(CustomMigrationTargetView.takeMeThereNotification).click();
+            cy.get("article", { timeout: 12 * SEC }).should("contain", target.name);
+
+            const newName = data.getRandomWord(8);
+            const newRules: CMTManualOrigin = {
+                type: CustomMigrationTargetOriginType.Manual,
+                rulesetPath: "xml/javax-package-custom.windup.xml",
+            };
+
+            target.edit({
+                name: newName,
+                rulesOrigin: newRules,
+            });
+            cy.wait("@putRule");
+            cy.get("article", { timeout: 12 * SEC }).should("contain", newName);
+            target.name = newName;
+            target.rulesOrigin = newRules;
+
+            target.delete();
+            cy.wait("@deleteRule");
+            cy.get("article", { timeout: 12 * SEC }).should("not.contain", target.name);
+        }
+    );
+
+    it(
+        "Create Custom Migration Target with repository fetching",
```

Comment:
Create Custom Migration Target with rules from repository",

Reply:
Solved in 32aa1a6660d5362ec0d81df4a0964a1286836d5c.